### PR TITLE
KAA-1631 Expand connection info

### PR DESCRIPTION
**Pull request by ohamov**
The patch adds the ability to find out that
an endpoint is connected to a server and the server
has accepted endpoint's credentials.

### CODE REVIEW

**Review by yverzun — APPROVED** [review]
LGTM

File: client/client-multi/client-cpp/kaa/EndpointConnectionInfo.hpp [hunk 1]
```diff
@@ -27,11 +27,17 @@ struct EndpointConnectionInfo {
     std::string endpointIp_; /**< The ip address of the endpoint. */
     std::string serverIp_; /**< The ip address of the server */
     ServerType serverType_; /**< The type of server, see @c ServerType */
+    /** Server confirmed that credentials are valid and approved the connection on an application level */
+    bool connectionAccepted_;
 
-    EndpointConnectionInfo(const std::string& endpointIp, const std::string& serverIp, const ServerType serverType)
+    EndpointConnectionInfo() = default;
```
Comment (RostakaGmfun) [inline client/client-multi/client-cpp/kaa/EndpointConnectionInfo.hpp:33]: I think it is a good idea to initialize members to some default values in this case

**Review by RostakaGmfun — APPROVED** [review]
Minor nitpick, otherwise LGTM

File: client/client-multi/client-cpp/kaa/EndpointConnectionInfo.hpp [hunk 2]
```diff
@@ -27,11 +27,17 @@ struct EndpointConnectionInfo {
     std::string endpointIp_; /**< The ip address of the endpoint. */
     std::string serverIp_; /**< The ip address of the server */
     ServerType serverType_; /**< The type of server, see @c ServerType */
+    /** Server confirmed that credentials are valid and approved the connection on an application level */
+    bool connectionAccepted_;
```
Comment (rasendubi) [inline client/client-multi/client-cpp/kaa/EndpointConnectionInfo.hpp:31]: Changes the structure layout, thus it's not binary compatible—can't go into the patch release.

**Review by rasendubi — CHANGES_REQUESTED** [review]

**Comment by rasendubi** [comment]
@GamovCoder consult Jira's `Fix Version` field before opening a PR. (Don't close this one—change the branch and re-push.)

File: client/client-multi/client-cpp/kaa/EndpointConnectionInfo.hpp (hunk 2 above)
Comment (GamovCoder) [inline client/client-multi/client-cpp/kaa/EndpointConnectionInfo.hpp]: Yep, I know. The another way to implement this leads to breaking interfaces. So, I've chosen this way.

File: client/client-multi/client-cpp/kaa/EndpointConnectionInfo.hpp (hunk 1 above)
Comment (ohamov) [reply client/client-multi/client-cpp/kaa/EndpointConnectionInfo.hpp:33]: They have default constructors

**Review by rasendubi — APPROVED** [review]

**Comment by rasendubi** [comment]
Jenkins failure is caused by KAA-1640.

**Pull request merged**